src/inputs/TextFieldBase.tsx

```diff
@@ -269,7 +271,9 @@ export function TextFieldBase<X extends Only<TextFieldXss, X>>(props: TextFieldB
       {/* Compound fields will handle their own error and helper text. Do not show error or helper text when 'readOnly' or disabled */}
       {!compound && !inputProps.disabled && !inputProps.readOnly && (
         <>
-          {errorMsg && !errorInTooltip && <ErrorMessage id={errorMessageId} errorMsg={errorMsg} {...tid.errorMsg} />}
+          {errorMsg && !errorInTooltip && !omitErrorMessage && (
+            <ErrorMessage id={errorMessageId} errorMsg={errorMsg} {...tid.errorMsg} />
+          )}
```

Comment:
**suggestion:** For screen reader purposes, it would be good to still have this error message in the DOM, but just hidden. Maybe we can do something like:
```
<ErrorMessage id={errorMessageId} errorMsg={errorMsg} hidden={hideErrorMessage} {...tid.errorMsg} />
```
```
import { VisuallyHidden } from "react-aria";

interface ErrorMessageProps {
  hidden?: boolean;
}

function ErrorMessage(props) {
  const { hidden = false } = props;
  const errorEl = ...;
  return hidden ? <VisuallyHidden>{errorEl}</VisuallyHidden> : errorEl;
}
```